pkg/registry/generic/etcd/etcd_test.go

```diff
@@ -263,78 +213,50 @@ func TestEtcdCreate(t *testing.T) {
-	nodeWithPodA := tools.EtcdResponseWithError{
-		R: &etcd.Response{
-			Node: &etcd.Node{
-				Value:         runtime.EncodeOrDie(testapi.Default.Codec(), podA),
-				ModifiedIndex: 1,
-				CreatedIndex:  1,
-			},
-		},
-		E: nil,
-	}
+	testContext := api.WithNamespace(api.NewContext(), "test")
+	server, registry := NewTestGenericEtcdRegistry(t)
+	defer server.Terminate(t)
 
-	emptyNode := tools.EtcdResponseWithError{
-		R: &etcd.Response{},
-		E: tools.EtcdErrorNotFound,
+	// create the object
+	objA, err := registry.Create(testContext, podA)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
 	}
 
-	testContext := api.WithNamespace(api.NewContext(), "test")
-
-	table := map[string]struct {
-		existing tools.EtcdResponseWithError
-		expect   tools.EtcdResponseWithError
-		toCreate runtime.Object
-		objOK    func(obj runtime.Object) bool
-		errOK    func(error) bool
-	}{
-		"normal": {
-			existing: emptyNode,
-			toCreate: podA,
-			objOK:    hasCreated(t, podA),
-			errOK:    func(err error) bool { return err == nil },
-		},
-		"preExisting": {
-			existing: nodeWithPodA,
-			expect:   nodeWithPodA,
-			toCreate: podB,
-			errOK:    errors.IsAlreadyExists,
-		},
+	// get the object
+	checkobj, err := registry.Get(testContext, podA.Name)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
 	}
 
-	for name, item := range table {
-		fakeClient, registry := NewTestGenericEtcdRegistry(t)
-		path := etcdtest.AddPrefix("pods/foo")
-		fakeClient.Data[path] = item.existing
-		obj, err := registry.Create(testContext, item.toCreate)
-		if !item.errOK(err) {
-			t.Errorf("%v: unexpected error: %v", name, err)
-		}
+	// verify objects are equal
+	if e, a := objA, checkobj; !api.Semantic.DeepDerivative(e, a) {
+		t.Errorf("Expected %#v, got %#v", e, a)
+	}
 
-		actual := fakeClient.Data[path]
-		if item.objOK != nil {
-			if !item.objOK(obj) {
-				t.Errorf("%v: unexpected returned: %v", name, obj)
-			}
-			actualObj, err := api.Scheme.Decode([]byte(actual.R.Node.Value))
-			if err != nil {
-				t.Errorf("unable to decode stored value for %#v", actual)
-				continue
-			}
-			if !item.objOK(actualObj) {
-				t.Errorf("%v: unexpected response: %v", name, actual)
-			}
-		} else {
-			if e, a := item.expect, actual; !api.Semantic.DeepDerivative(e, a) {
-				t.Errorf("%v:\n%s", name, util.ObjectDiff(e, a))
-			}
-		}
+	// now try to create the second pod
+	_, err = registry.Create(testContext, podB)
+	if !errors.IsAlreadyExists(err) {
+		t.Errorf("Unexpected error: %v", err)
 	}
 }
 
-func podCopy(in *api.Pod) *api.Pod {
-	out := *in
-	return &out
+func updateAndVerify(t *testing.T, ctx api.Context, registry *Etcd, pod *api.Pod) bool {
+	obj, _, err := registry.Update(ctx, pod)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+		return false
+	}
+	checkObj, err := registry.Get(ctx, pod.Name)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+		return false
+	}
+	if e, a := obj, checkObj; !api.Semantic.DeepDerivative(e, a) {
```

Comment:
As above, can you please add a comment that we are using DeepDerivative, because of resource versions and defaulting (and it has to be (e, a) not (a,e))?